torch/csrc/api/include/torch/nn/functional/pooling.h

```diff
@@ -427,15 +427,14 @@ inline Tensor max_unpool1d(
 } // namespace detail
 
 inline Tensor max_unpool1d(const Tensor& input, const Tensor& indices,
-    MaxUnpool1dFuncOptions options,
-    const c10::optional<IntArrayRef>& output_size = c10::nullopt) {
+    MaxUnpool1dFuncOptions options) {
```

Comment:
I don't mind but I still can't imagine any scenario when pass-by-value is better than pass-by-const-ref

Reply:
Yeah I agreed, I think an alternative is to make all functional options pass-by-const-ref except for `NormalizeFuncOptions`, because we need it to be mutable in order to store output in the `out` Tensor. Currently all functional options are pass-by-value mostly for consistency, and we can revisit this later if perf is a concern.

Reply:
EDIT: I will change all options to pass-by-const-ref except for `NormalizeFuncOptions`, so that we don't need to worry about this later.

Reply:
Fixed